conans/client/installer.py

```diff
@@ -197,6 +197,11 @@ def _build(self, nodes_by_level, skip_private_nodes, deps_graph):
         nodes_to_process = self._get_nodes(nodes_by_level, skip_private_nodes)
 
         for conan_ref, package_id, conan_file, build_needed in nodes_to_process:
+
+            if conan_ref and (not conan_file.version or not conan_file.name):
```

Comment:
I don't think this is enough for all cases. It should be injected at load time. also conan_file.version might not be defined at all, not None, but raise Exception.